### PR TITLE
Add map_string_as_varchar session property in ClickHouse

### PR DESCRIPTION
**Pull request by ebyhr**
Relates to #7102

Docs will be handled in #7099

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #10363

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java [hunk 1]
```diff
@@ -252,6 +270,17 @@ public void testClickHouseFixedString()
...
+        // Set map_string_as_varchar session property as true
+        SqlDataTypeTest.create()
+                // plain
+                .addRoundTrip("FixedString(10)", "'c12345678b'", VARCHAR, "CAST('c12345678b' AS varchar)")
+                .addRoundTrip("FixedString(10)", "'c123'", VARCHAR, "CAST('c123\0\0\0\0\0\0' AS varchar)")
+                // nullable
+                .addRoundTrip("Nullable(FixedString(10))", "NULL", VARCHAR, "CAST(NULL AS varchar)")
+                .addRoundTrip("Nullable(FixedString(10))", "'c12345678b'", VARCHAR, "CAST('c12345678b' AS varchar)")
+                .addRoundTrip("Nullable(FixedString(10))", "'c123'", VARCHAR, "CAST('c123\0\0\0\0\0\0' AS varchar)")
+                .execute(getQueryRunner(), mapStringAsVarcharSession(), clickhouseCreateAndInsert("tpch.test_fixed_string"));
```
Comment (findepi) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java:283]: Do we want to have a test when we insert without `mapStringAsVarcharSession` and read with it, and vice versa?

**Review by findepi — APPROVED** [review]

**Comment by hashhar** [comment]
Question : Could the same effect be achieved via `jdbc-types-mapped-to-varchar`?

**Comment by ebyhr** [comment]
Yes if users specify the name including the length.
* `map-string-as-varchar` map all `FixedString` to varchar regardless of the length. 
* `jdbc-types-mapped-to-varchar` expects the entire type name. e.g. `FixedString(5)`

**Pull request merged**